src/ducks/modules/network.js

```diff
@@ -32,28 +37,57 @@ function edgeExists(edges, edge) {
   );
 }
 
-function getNodesWithBatchAdd(oldNodes, newNodes, additionalAttributes) {
-  const withAttrs = newNode => ({ ...additionalAttributes, [NodePK]: uuidv4(), ...newNode });
-  return oldNodes.concat(newNodes.map(withAttrs));
+/**
+ * existingNodes - Existing network.nodes
+ * netNodes - nodes to be added to the network
+ * additionalAttributes - node model properties
+*/
+function getNodesWithBatchAdd(existingNodes, newNodes, nodeAttributeData) {
+  // Create a function to create a UUID and merge node attributes
+  const withModelandAttributeData = newNode => ({
+    [NodePrimaryKeyProperty]: uuidv4(),
+    ...newNode, // second to allow existing UUID to be overwritten
+    [NodeAttributesProperty]: {
+      ...newNode[NodeAttributesProperty],
+      ...nodeAttributeData,
+    },
+  });
+
+  return existingNodes.concat(newNodes.map(withModelandAttributeData));
 }
 
-function getUpdatedNodes(nodes, updatedNode, full) {
+/**
+ * @param {array} nodes - an array of objects representing nodes
+ * @param {object} updatedNode - object representing the node to be updated and its new properties
+ * @param {boolean} full - if the entire node should be overwritten
+ */
+function getUpdatedNodes(nodes, updatedNode, nodeAttributeData) {
```

Comment:
Is loosing the ability to do a replace operation ('full'), going to have any side-effects?